main.go

```diff
@@ -783,6 +822,43 @@ var DevCopyCommand = cli.Command{
 	},
 }
 
+func genLockDeps(pkg *Package, deps map[string]gx.LockDep, done map[string]bool, ignoreConflict bool) error {
+	for _, dep := range pkg.Dependencies {
+		if done[dep.Hash] {
+			continue
+		}
+
+		done[dep.Hash] = true
+
+		var cpkg Package
+		err := gx.LoadPackage(&cpkg, pkg.Language, dep.Hash)
+		if err != nil {
+			if os.IsNotExist(err) {
+				VLog("LoadPackage error: ", err)
+				return fmt.Errorf("package %s (%s) not found", dep.Name, dep.Hash)
+			}
+			return err
+		}
+
+		ref := fmt.Sprintf("/ipfs/%s/%s", dep.Hash, dep.Name)
```

Comment:
This seems to be the actual core of function where we are transforming a `Dependency` to a `Ref`, could we add a comment to highlight this? (And even better, make it into a function.)